src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -185,26 +192,58 @@ private void drawLineWithTranslate(Graphics2D graphics, Line2D line, double xOff
    * 
    * @param routeDescription {@linkplain RouteDescription} containing the Route information
    * @param mapData {@linkplain MapData} Object containing Information about the Map Coordinates
+   * @param width The width of the Map
+   * @param height The height of the Map
    * @return The {@linkplain Point} array specified by the {@linkplain RouteDescription} and {@linkplain MapData}
    *         objects
    */
-  protected Point[] getRoutePoints(RouteDescription routeDescription, MapData mapData) {
+  protected Point[] getRoutePoints(RouteDescription routeDescription, MapData mapData, int xOffset, int yOffset,
+      int width, int height) {
     final List<Territory> territories = routeDescription.getRoute().getAllTerritories();
     final int numTerritories = territories.size();
     final Point[] points = new Point[numTerritories];
     for (int i = 0; i < numTerritories; i++) {
-      points[i] = mapData.getCenter(territories.get(i));
+      points[i] = getPointOnMap(mapData.getCenter(territories.get(i)), xOffset, yOffset, width, height);
```

Comment:
Not very ideal to have 4 integer params, they can easily be written out of order. the 'width' and 'height' values should fit pretty naturally into a Dimension object, which at least reduces one of the params.